qdlt/qdltpluginmanager.cpp

```diff
@@ -134,108 +151,240 @@ QStringList QDltPluginManager::loadPluginsPath(QDir &dir)
 
 void QDltPluginManager::loadConfig(QString pluginName,QString filename)
 {
+    pMutex_pluginList->lock();
     for(int num=0;num<plugins.size();num++)
     {
         QDltPlugin *plugin = plugins[num];
         if(plugin->getName()==pluginName)
             plugin->setFilename(filename);
     }
+    pMutex_pluginList->unlock();
 }
 
 
 void QDltPluginManager::decodeMsg(QDltMsg &msg, int triggeredByUser)
 {
+    pMutex_pluginList->lock();
     for(int num=0;num<plugins.size();num++)
     {
         QDltPlugin *plugin = plugins[num];
 
         if(plugin->decodeMsg(msg,triggeredByUser))
             break;
-
     }
+    pMutex_pluginList->unlock();
 }
 
 QDltPlugin* QDltPluginManager::findPlugin(QString &name)
 {
+    QDltPlugin *plugin = nullptr;
+
+    pMutex_pluginList->lock();
     for(int num=0;num<plugins.size();num++)
     {
-        QDltPlugin *plugin = plugins[num];
+        if(plugins[num]->getName()==name)
+        {
+            plugin = plugins[num];
+            break;
+        }
+    }
+    pMutex_pluginList->unlock();
+
+    return plugin;
```

Comment:
this is unsafe, or? does the mutex only protect the order of elements? or can items as well be changed/modified/deleted?

Reply:
the mutex only protects the order. The elements still can be changed while decoding (as before)